cunumeric/module.py

```diff
@@ -2372,6 +2426,198 @@ def diag_indices_from(arr):
     return diag_indices(arr.shape[0], arr.ndim)
 
 
+def mask_indices(n, mask_func, k=0):
```

Comment:
I'd remove this implementation and let us revisit this later. this naive implementation will always force the input cuNumeric ndarray to be materialized on the host memory so that you can pass it to a random python function, which is far from what we want. (it's also not true that it supports multi-node multi-GPU execution.)